OPHD/Technology/TechnologyCatalog.cpp

```diff
@@ -98,30 +98,25 @@ void TechnologyCatalog::readTechnologiesInCategory(const std::string& categoryNa
 {
 	for (auto technology = category.firstChildElement(); technology; technology = technology->nextSiblingElement())
 	{
-		const auto attributes = NAS2D::attributesToDictionary(*technology);
-		const int id = attributes.get<int>("id");
+		Technology tech = readTechnology(*technology);
 
 		const auto& technologies = mCategories[categoryName];
-		const auto it = std::find_if(technologies.begin(), technologies.end(), [id](const Technology& tech) { return tech.id == id; });
+		const auto it = std::find_if(technologies.begin(), technologies.end(), [tech](const Technology& technology) { return technology.id == tech.id; });
 		if (it != technologies.end())
 		{
-			throw std::runtime_error("TechnologyReader: Technology ID redefinition '" + std::to_string(id) +
+			throw std::runtime_error("TechnologyReader: Technology ID redefinition '" + std::to_string(tech.id) +
 				"' at (" + std::to_string(technology->row()) + ", " + std::to_string(technology->column()) + ")");
 		}
-
-		readTechnology(categoryName, *technology, attributes);

```

Comment:
Looks like a small oversight with this change. The `tech` value isn't being stored anywhere after reading it. Previously that was being done inside of `readTechnology`, which was modified to instead return the read value.

Aside from that, this nicely decouples the code.

Edit: I see that was corrected in a later commit.

Reply:
Yeah, I didn't actually test the code after compiling and realized when testing something else that the technologies themselves weren't getting saved. Wherps.